### ISSUE TITLE
I'm getting results that are different from the MB website

### ISSUE THREAD

**Issue by barrars** [Borewit/musicbrainz-api#124]
First, thank you for writting this module!!

So I'm trying to recreate this query:
[https://musicbrainz.org/search?query=Beatles%2CThe+Lucy+In+The+Sky+With+Diamonds&type=recording](https://musicbrainz.org/search?query=Beatles%2CThe+Lucy+In+The+Sky+With+Diamonds&type=recording)

with this code:
```js
  const res = await mbApi.search('recording', 'Beatles,The Lucy In The Sky With Diamonds ')
```
The url results includes The Beatles in the first several results (which is what I'm looking for).
The mbApi search doesnt have any Beatles until the 24th result.

Am I doing this correctly?

**Comment by BradNut** [comment]
@Borewit, I am seeing the same thing when trying to get releases for an artist. Results for artist "Baroness" should show all release on the website shown here: https://musicbrainz.org/artist/eeb41a1e-4326-4d04-8c47-0f564ceecd68

Instead the results only return releases up to 2012 "Yellow & Green". Is there some limitation in the API from Musicbrainz's side?

**Comment by Borewit** [comment]
Thank you for reminding me to this question @BradNut. Apologies @barrars.

Yeah, [there is a limitation to this API](https://musicbrainz.org/doc/MusicBrainz_API#Subqueries). It will not give you all the results at once:

> Note that the number of linked entities returned is always limited to 25. If you need the remaining results, you will have to perform a browse request.


`result.count` provides you the number of items found.

The following code will probably give you the next 25 search results
```js
 const res = await mbApi.search('recording', 'Beatles,The Lucy In The Sky With Diamonds', {offset: 25});
```

**Comment by BradNut** [comment]
@Borewit Thank you for responding. It makes sense now given Musicbrainz limit of 25. In my case I am directly calling `getArtist()` passing an ID and subqueries for the releases part. Is there a way to get offsets of an Artist's releases?

My current query:
```
const searchResult: IArtist = await mbApi.getArtist(artist?.musicBrainzId, [
      'artist-rels',
      'releases',
      'recordings',
      'url-rels',
    ]);
```

**Comment by Borewit** [comment]
> Is there a way to get offsets of an Artist's releases?

I don't think there is straight forward way from there to fetch the rest. You probably have to an explicit search for releases from that artist.

```js
  // Get releases from artist eeb41a1e-4326-4d04-8c47-0f564ceecd68 starting at offset=25  
  const result = await mbApi.searchRelease({artist: 'eeb41a1e-4326-4d04-8c47-0f564ceecd68'}, 25);
```
Maybe that nicely lines up. not sure, give it a try.

**Comment by BradNut** [comment]
Doing this results in the following query being sent by your method
```
response = await got.get('ws/2' + relUrl, {
        searchParams: query,
        responseType: 'json',
        ...this.options
      });
```
Results in URL: https://musicbrainz.org/ws/2/release/?query=artist%3A%22eeb41a1e-4326-4d04-8c47-0f564ceecd68%22&offset=0&limit=25&fmt=json%27,

Code:
```
let offset = 0;
    let releaseList: IReleaseList = await mbApi.searchRelease(
      { artist: artist?.musicBrainzId },
      offset,
      25
    );
```

Shouldn't the resulting url that goes out be something like `https://musicbrainz.org/ws/2/release/?artist=eeb41a1e-4326-4d04-8c47-0f564ceecd68&offset=0&limit=25&fmt=json`?

And looks like the resulting URL has extra `%27,` at the end. Thoughts?

**Comment by BradNut** [comment]
@Borewit For some reason it wasn't working but I did get something to work for just searching releases by `arid`. Not sure if it's as good or efficient...?

```
let releaseList = await mbApi.search(
      'release',
      `arid:${artist?.musicBrainzId}`,
      offset,
      25
    );
```

**Comment by Borewit** [comment]
> Shouldn't the resulting url that goes out be something like https://musicbrainz.org/ws/2/release/?artist=eeb41a1e-4326-4d04-8c47-0f564ceecd68&offset=0&limit=25&fmt=json?

That is using the [browse](https://musicbrainz.org/doc/MusicBrainz_API#Browse) [lookup](https://musicbrainz.org/doc/MusicBrainz_API#Lookups) instead of the [query](https://musicbrainz.org/doc/MusicBrainz_API/Search) [lookup](https://musicbrainz.org/doc/MusicBrainz_API/Lookups). I will improve the module so you can pick the one you prefer.

> And looks like the resulting URL has extra %27, at the end. Thoughts?

Weird, I actually did not find that extra quote.

**Comment by BradNut** [comment]
@Borewit Updated to the new version and I can successfully search releases by ARID. Seems a bit cleaner now this way as well.

Does this look like the correct way to do this? Where I increment offset plus 25 when I need it. Just wanted to check.
```
let releaseList: IReleaseList = await mbApi.searchRelease({
  query: `arid:${artist?.musicBrainzId}`,
  offset,
  limit: 25,
});
```

**Comment by BradNut** [comment]
@Borewit One more question regarding `inc` params on search requests. I tried passing an array of strings for what I want included but I end up getting the following Type error:
```
TypeError: The `searchParams` value 'release-groups,media,label-rels' must be a string, number, boolean or null
    at validateSearchParameters (/home/bshellnu/projects/musicle/backend/node_modules/got/dist/source/core/index.js:67:19)
    at normalizeArguments (/home/bshellnu/projects/musicle/backend/node_modules/got/dist/source/core/index.js:446:21)
    at got (/home/bshellnu/projects/musicle/backend/node_modules/got/dist/source/create.js:112:39)
    at Function.got.<computed> [as get] (/home/bshellnu/projects/musicle/backend/node_modules/got/dist/source/create.js:224:42)
    at MusicBrainzApi.restGet (/home/bshellnu/projects/musicle/backend/node_modules/musicbrainz-api/lib/musicbrainz-api.js:107:44)
```

Code:
```
releaseList = await mbApi.searchRelease({
        query: `arid:${artist?.musicBrainzId}`,
        inc: ['release-groups', 'media', 'label-rels'],
        offset,
        limit: 25,
      });
```

**Comment by Borewit** [comment]
> but I end up getting the following Type error

That was a [bug](https://github.com/Borewit/musicbrainz-api/issues/390), should be fixed in [v0.7.2](https://github.com/Borewit/musicbrainz-api/releases/tag/v0.7.2).

Regarding your code:
```js
let releaseList: IReleaseList = await mbApi.searchRelease({
  query: `arid:${artist?.musicBrainzId}`,
  offset,
  limit: 25
});
```
looks fine, you could also provide an object to the query. Multiple properties will be combined to an AND query.
```js
let releaseList: IReleaseList = await mbApi.searchRelease({
  query: {arid: artist?.musicBrainzId},
  offset,
  limit: 25
});
```
or use the _browse_ API, which has a slightly different result structure:
```js
let releaseList: IReleaseList = await mbApi.searchRelease({
  artist: artist.musicBrainzId,
  offset,
  limit: 25
});
```
I have no strong preference.
